cli/agent_test.go

```diff
@@ -153,17 +153,18 @@ func TestWorkspaceAgent(t *testing.T) {
 			context.WithValue(inv.Context(), "aws-client", metadataClient),
 		)
 		clitest.Start(t, inv)
+		ctx := inv.Context()
 		coderdtest.AwaitWorkspaceAgents(t, client, workspace.ID)
-		workspace, err := client.Workspace(inv.Context(), workspace.ID)
+		workspace, err := client.Workspace(ctx, workspace.ID)
 		require.NoError(t, err)
 		resources := workspace.LatestBuild.Resources
 		if assert.NotEmpty(t, resources) && assert.NotEmpty(t, resources[0].Agents) {
 			assert.NotEmpty(t, resources[0].Agents[0].Version)
 		}
-		dialer, err := client.DialWorkspaceAgent(inv.Context(), resources[0].Agents[0].ID, nil)
+		dialer, err := client.DialWorkspaceAgent(ctx, resources[0].Agents[0].ID, nil)
 		require.NoError(t, err)
 		defer dialer.Close()
-		require.True(t, dialer.AwaitReachable(context.Background()))
```

Comment:
I think that this is a second PR fixing `context.Background()`. I'm curious if there is a lint-way to prevent this dev mistake.

Reply:
It would be unfortunate to ban `context.Background()` entirely since there are valid use-cases. But most of our tests should try to have a timeout so maybe? Ruleguard (`scripts/rules.go`) would probably be the place to put it, but it's a bit finicky to get right.

In theory, we have a lot of "debt" here:

```
❯ rg 'context\.Background\(\)' **/*_test.go | grep -E -v 'context\.With(Timeout|Deadline|Value)\(' | wc -l
451
```